File: python_playground/selection.py
def sort(list_to_sort: list[int]) -> list[int]:
    for i in range(0, len(list_to_sort) - 1):
        min_index = i
        for k in range(i + 1, len(list_to_sort)):
            if list_to_sort[k] < list_to_sort[min_index]:
                min_index = k
        if min_index != i:
            list_to_sort[i], list_to_sort[min_index] = list_to_sort[min_index], list_to_sort[i]
    return list_to_sort

File: algo_visualizer/src/algos/selection.ts
import type SortAlgorithm from "./SortAlgorithm";
import { BigO } from "./SortAlgorithm";

export default class Selection implements SortAlgorithm {
  readonly name = "Selection Sort";
  readonly expectedBigO: BigO = BigO.N_SQUARED;

  sort = (listToSort: number[]): number[] => {
    for (let i = 0; i < listToSort.length - 1; i++) {
      let minIndex = i;
      for (let k = i + 1; k < listToSort.length; k++) {
        if (listToSort[k] < listToSort[minIndex]) {
          minIndex = k;
        }
      }
      if (minIndex !== i) {
        [listToSort[i], listToSort[minIndex]] = [listToSort[minIndex], listToSort[i]];
      }
    }
    return listToSort;
  };
}
